contracts/RelayHub.sol

```diff
@@ -8,15 +8,23 @@ import "@0x/contracts-utils/contracts/src/LibBytes.sol";
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 
 contract RelayHub is IRelayHub {
-    // Minimum values for stake
-    uint256 constant public minimumStake = 0.1 ether;
-    uint256 constant public minimumUnstakeDelay = 0;
+    // Minimum stake a relay can have. An attack to the network will never cost less than half this value.
+    uint256 constant public minimumStake = 1 ether;
 
-    // Minimum balance required for a relay to register or re-register
+    // Minimum unstake delay. A relay needs to wait for this time to elapse after deregistering to retrieve its stake.
+    uint256 constant public minimumUnstakeDelay = 1 weeks;
+    // Maximum unstake delay. Prevents relays from locking their funds into the RelayHub for too long.
+    uint256 constant public maximumUnstakeDelay = 12 weeks;
+
+    // Minimum balance required for a relay to register or re-register. Prevents user error in registering a relay that
+    // will not be able to immediatly start serving requests.
     uint256 constant public minimumRelayBalance = 0.1 ether;
 
-    // Maximum funds that can be deposited at once. Prevents user errors by disallowing large deposits.
-    uint256 constant public maximumDeposit = 2 ether;
+    // Maximum funds that can be deposited at once. Prevents user error by disallowing large deposits.
+    uint256 constant public maximumRecipientDeposit = 2 ether;
+
+    // Maximum amount that can be staked at once. Prevents user error by disallowing large deposits.
+    uint256 constant public maximumStakeDeposit = 5 ether;
```

Comment:
why prevent large staking? 
for recipients, it makes sense since:
1. 2 ether allows huge # of transactions, 
2. security-wise, deposit means nothing, and 
3. RR Implementation might forget to add a "withdraw" implementation, making this deposit "non-refundable" for the foreseeable future..

But for relay,  staking means the trust a user can put on it: I might prefer a 100-ether-staked relay for high-valued transactions.  Also, there is always a way to unstake (even if with some delay) - so there is no real reason to have such a limit.

Reply:
To clarify, this doesn't prevent large stakes, but rather caps how much a stake can be increased by in a single transaction. Similarly, a user may deposit as much ether as they want in a contract, but they have to do it over multiple calls to `deposit`. 

Limiting this value (just like `unstakeDelay` is also capped) seems like a good idea from a usability perspective: it limits the severity of user mistakes (e.g. you won't stake all of your balance due to a typo).

Doing a couple transactions to achieve a stake higher than 5 ETH doesn't seem like a big issue imo (considering 5 ETH is already ~1.5k USD), but if we agree it makes sense to have this cap, we can increase this value is you think the limit is low.

Reply:
This limit of a kind of saying that we don't think you should stake more.
Yes, it makes little sense to have a cap, but it should be big, e.g 100
UI tool to deposit it might add warning are 'are you sure' messages. It's not the contact's job.

Reply:
I removed `maximumStakeDeposit`: I agree with your assessment that there is a mechanism to retrieve it anyway, so this is more of an UI concern and not the contract's.